tests/installation/partitioning/edit_mount_by.pm

```diff
@@ -0,0 +1,52 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+# Summary: Edit suggested partitioning proposal and change the mounting option
+# for some partitions, according to test_data.
+# test_data syntax example:
+#
+# test_data:
+#   disks:
+#     - name: sda
+#       partitions:
+#         - name: sda1
+#           fstab_options:
+#             mount_by: UUID
+#         - name: sda2
+#           fstab_options:
+#             mount_by: "Device name"
+#
+# Maintainer: QA SLE YaST team <qa-sle-yast@suse.de>
+
+use strict;
+use warnings;
+use parent 'y2_installbase';
+use testapi;
+use version_utils ':VERSION';
+use scheduler 'get_test_suite_data';
+
+sub run {
+    my $test_data   = get_test_suite_data();
+    my $partitioner = $testapi::distri->get_expert_partitioner();
+    $partitioner->run_expert_partitioner('current');
+    foreach my $disk (@{$test_data->{disks}}) {
+        foreach my $partition (@{$disk->{partitions}}) {
+            if ($partition->{name}) {
+                record_info("Edit $partition->{name}", "Set $partition->{name} mount_by : $partition->{mount_by}");
+                $partitioner->edit_partition_gpt({
```

Comment:
Hmm, seems that I miss something, but from the code I see that new method you have implemented (`edit_fstab_options`) is not called from `edit_partition_gpt`. Do we miss some commit?
I also wonder about module itself, as not it can be used to edit any partition and is completely data_driven. Feels like we can call it `edit_partition_gpt` and use test data to define what exactly will be changed.
Also, for this scenario we can consider adding all partitions, as now it seems that we rely on the proposal which might change over time and editing will fail.

Reply:
The edit_fstab_options is called by edit_partition_not_encrypted, which is called by edit_partition_gpt. You are right about the module, it basically is the same as edit_partition_encrypt, which is almost the same as modify_existing_partition. I will try to combine those 3 to one, more generic, the changes shouldn't be significant. About the option to add all partitions from the beginning, like setup_lvm, I think it's better coverage to use both ways in installing tests. In the specific scenario, if the suggested partitioning changes, we will probably just need to make small changes in the test data. WDYT?

Reply:
Thanks for pointing me to the right place. Good catch that we have 3 modules doing similar thing. Definitely it's worth covering both, setting `mount by` option and changing it. However, for changing it, we need to be sure that previous value is set properly (not only in the UI), so I would start with new partitions. And then we could add scenario for the Expert Partitioner in the installed system, or use disk from this test as an input.

Reply:
I went for new partitions, but I think that a follow up ticket to cover the scenario where we edit the partitioning, instead of a new would be a bit too much, no?

Reply:
I think it's worth filing in any case, and then we see when we can act on it. Otherwise, we will forget about this idea.